src/cursor/parse_cursor.rs

```diff
@@ -0,0 +1,567 @@
+            0x00, 0x00, 0x00, 0x00, // height 0
+            0x00, 0x00, 0x00, 0x00, // x_hot 0
+            0x00, 0x00, 0x00, 0x00, // y_hot 0
+            0x00, 0x00, 0x00, 0x00, // delay 0
+        ];
+        match Image::read(&mut Cursor::new(&data[..]), IMAGE_TYPE, 4) {
+            Err(Error::ImageTooLarge) => {}
+            r => panic!("Unexpected result {:?}", r),
+        }
+    }
+
+    #[test]
+    fn read_image_too_large_height() {
+        let data = [
+            0x00, 0x00, 0x00, 0x00, // header(?)
+            0x02, 0x00, 0xfd, 0xff, // IMAGE_TYPE
+            0x04, 0x00, 0x00, 0x00, // size 4
+            0x00, 0x00, 0x00, 0x00, // version(?)
+            0x00, 0x00, 0x00, 0x00, // width 0
+            0x00, 0x80, 0x00, 0x00, // height IMAGE_MAX_SIZE + 1
+            0x00, 0x00, 0x00, 0x00, // x_hot 0
+            0x00, 0x00, 0x00, 0x00, // y_hot 0
+            0x00, 0x00, 0x00, 0x00, // delay 0
+        ];
+        match Image::read(&mut Cursor::new(&data[..]), IMAGE_TYPE, 4) {
+            Err(Error::ImageTooLarge) => {}
+            r => panic!("Unexpected result {:?}", r),
+        }
+    }
+
+    #[test]
+    fn read_image_too_short() {
+        let data = [];
+        match Image::read(&mut Cursor::new(&data[..]), IMAGE_TYPE, 4) {
+            Err(Error::IO(e)) if e.kind() == ErrorKind::UnexpectedEof => {}
+            r => panic!("Unexpected result {:?}", r),
+        }
+    }
+
+    #[test]
+    fn find_best_size_empty_input() {
+        let res = find_best_size(&[], 42);
+        match res {
+            Err(Error::NoImages) => {}
+            r => panic!("Unexpected result {:?}", r),
+        }
+    }
+
+    #[test]
+    fn find_best_size_one_input() {
+        let input = [TocEntry {
+            kind: IMAGE_TYPE,
+            size: 42,
+            position: 42,
+        }];
+        assert_eq!(42, find_best_size(&input, 10).unwrap());
+    }
+
+    #[test]
+    fn find_best_size_selects_better() {
+        let input = [
+            TocEntry {
+                kind: IMAGE_TYPE,
+                size: 42,
+                position: 42,
+            },
+            TocEntry {
+                kind: IMAGE_TYPE,
+                size: 32,
+                position: 42,
+            },
+            TocEntry {
+                kind: IMAGE_TYPE,
+                size: 3,
+                position: 42,
+            },
+            TocEntry {
+                kind: IMAGE_TYPE,
+                size: 22,
+                position: 42,
+            },
+            TocEntry {
+                kind: 0,
+                size: 10,
+                position: 42,
+            },
+        ];
+        assert_eq!(3, find_best_size(&input, 10).unwrap());
+    }
+
+    #[test]
+    fn parse_cursor_too_short() {
+        let data = [];
+        match parse_cursor(&mut Cursor::new(&data[..]), 10) {
+            Err(Error::IO(e)) if e.kind() == ErrorKind::UnexpectedEof => {}
```

Comment:
Bind `e` by-reference to make 1.37 happy

```suggestion
            Err(Error::IO(ref e)) if e.kind() == ErrorKind::UnexpectedEof => {}
```